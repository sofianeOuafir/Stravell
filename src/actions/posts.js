import database from "../firebase/firebase";

const addPost = post => ({
  type: "ADD_POST",
  post
});

const startAddPost = (postData = {}) => {
  return (dispatch, getState) => {
    const { uid, userName, userPhotoURL } = getState().auth;
    const { title, body, description, image, createdAt, updatedAt } = postData;
    const post = {
      uid,
      userName,
      userPhotoURL,
      title,
      description,
      image,
      body,
      createdAt,
      updatedAt
    };
    return database
      .ref("posts")
      .push(post)
      .then(ref => {
        dispatch(
          addPost({
            id: ref.key,
            ...post
          })
        );
      })
      .catch(e => console.log(e));
  };
};

const editPost = ({ id, updates }) => ({
  type: "EDIT_POST",
  id,
  updates
});

const startEditPost = ({ id, updates }) => {
  return dispatch => {
    database
      .ref(`posts/${id}`)
      .update(updates)
      .then(ref => {
        dispatch(editPost({ id, updates }));
      });
  };
};

const setPosts = posts => ({
  type: "SET_POSTS",
  posts
});

const startSetPosts = () => {
  return dispatch => {
    return new Promise((resolve, reject) => {
      let posts = [];
      database
      .ref("posts")
      .orderByChild("createdAt")
      .on("value", snapshot => {
        snapshot.forEach(snapshotChild => {
          posts.push({
            id: snapshotChild.key,
            ...snapshotChild.val()
          });
        });
        posts = posts.reverse();
        dispatch(setPosts(posts));
        resolve();
      });
      
    })
  };
};

export { startEditPost, startAddPost, startSetPosts };
