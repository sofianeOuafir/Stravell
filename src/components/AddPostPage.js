import React from "react";
import { connect } from "react-redux";
import { withRouter } from "next/router";
import { slugify } from "underscore.string";

import PostForm from "./PostForm";
import PageHeader from "./PageHeader";
import {
  ADD_POST_PAGE_TITLE,
  ADD_POST_PAGE_DESCRIPTION
} from "./../constants/constants";
import Layout from "./Layout";
import BreadCrumb from "./Breadcrumb";
import { addPost } from "./../queries/post";

class AddPostPage extends React.Component {
  onSubmit = ({ post, country, user, place, region }) => {
    const { addPost } = this.props;
    addPost({ post, country, user, place, region }).then(() => {
      this.props.router.push(
        `/dashboard?uid=${this.props.uid}`,
        `/dashboard/${slugify(this.props.userName)}/${this.props.uid}`
      );
    });
  };

  render() {
    const { userName, uid } = this.props;
    const breadcrumbLinks = [
      { href: "/", text: "Home" },
      {
        href: `/dashboard?uid=${uid}`,
        as: `/dashboard/${slugify(userName)}/${uid}`,
        text: "Dashboard"
      },
      {
        href: `/create`,
        as: "/p/create",
        text: "Create Post",
        active: true
      }
    ];
    return (
      <Layout
        title={ADD_POST_PAGE_TITLE}
        description={ADD_POST_PAGE_DESCRIPTION}
      >
        <PageHeader title="Create Post" withSocialShareButtons={false} />
        <div className="content-container">
          <div className="mb1">
            <BreadCrumb links={breadcrumbLinks} />
          </div>

          <PostForm onSubmit={this.onSubmit} />
        </div>
      </Layout>
    );
  }
}

AddPostPage.getInitialProps = async function({ req, reduxStore, res }) {
  let authorised = false;
  if (req && req.session.decodedToken) {
    const user = req.session.decodedToken;
    if (user) {
      authorised = true;
    }
  } else {
    if (reduxStore.getState().auth.uid) {
      authorised = true;
    }
  }

  if (authorised) {
    return {};
  } else {
    if (res) {
      res.writeHead(302, {
        Location: "/"
      });
      res.end();
    } else {
      Router.push("/");
    }
  }
};

const mapDispatchToProps = () => ({
  addPost: ({ post, country, user, place, region }) => {
    return addPost({ post, country, user, place, region });
  }
});

const mapStateToProps = state => ({
  uid: state.auth.uid,
  userName: state.auth.userName
});

export { AddPostPage };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(AddPostPage));
