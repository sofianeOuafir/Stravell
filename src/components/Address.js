import React from "react";
import { MdMyLocation } from "react-icons/md";
import Link from "next/link";
import { slugify } from "underscore.string";

const Address = ({ iconClassName = "", addressClassName = "", address, placeId }) => {
  return (
    <Link
      href={`/place?id=${placeId}`}
      as={`/place/${slugify(address)}/${placeId}`}
    >
      <a className="flex align-items--center">
        <MdMyLocation className={iconClassName} />
        <span className={addressClassName}>
          {address}
        </span>
      </a>
    </Link>
  );
};

export default Address;
