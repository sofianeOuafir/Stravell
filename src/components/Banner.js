import React from "react";
import { IoIosSearch, IoIosAirplane } from "react-icons/io";
import Typed from "react-typed";
import { Link as ScrollLink } from "react-scroll";

import SocialShareButtons from "./SocialShareButtons";
import Image from './Image';
class Banner extends React.Component {
  render() {
    const { imageUrl, place, placeSuggestions } = this.props;
    return (
      <div className="banner__image-container">
        <div className="content-container">
          <div className="banner__text-container" />
          <div className="banner__text">
            <h1 className="banner__website-name">
              Stravell {place ? `| ${place}` : ""}
            </h1>
            <span className="banner__description">
              A search engine for travel articles. <br /> It makes it easy for you
              to find travel articles about thousands of places
              <br />
              around the world written by some of the best travel bloggers
              worldwide.
            </span>

          </div>
          <div className="banner__typing-container" />
          <div className="banner__typing flex align-items--center">
            <IoIosSearch className="mr1" />
            <Typed
              strings={placeSuggestions}
              typeSpeed={100}
              backSpeed={100}
              loop
            />
          </div>
          <div className="banner__get_started-container">
            <ScrollLink
              href="#"
              className="button button--large button--warm-peach"
              to="searchBarInput"
              smooth="true"
              duration={600}
              offset={-100}
            >
              <div className="flex align-items--center">
                <IoIosAirplane className="mr1" /> <span>Get Started</span>
              </div>
            </ScrollLink>
          </div>

          <div className="banner__social_buttons">
            <SocialShareButtons />
          </div>
        </div>
        <Image className="fullwidth" src={imageUrl} alt="" loadingImageWidth={1026} loadingImageHeight={405} />
      </div>
    );
  }
}

export default Banner;
