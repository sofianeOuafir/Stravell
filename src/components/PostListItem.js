import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

const PostListItem = ({ post }) => {
  let jsx = (
    <div>
      <p>{post.title}</p>
      <p>{post.body}</p>
    </div>
  );

  {
    isOwnedByAuthenticatedUser
      ? ( { jsx = <Link to={`/posts/edit/${post.id}`}>jsx</Link> })
      : jsx;
  }

  return jsx;
};

const mapStateToProps = (state, props) => ({
  isOwnedByAuthenticatedUser: state.auth.uid === props.post.uid
});

export default connect()(PostListItem);
