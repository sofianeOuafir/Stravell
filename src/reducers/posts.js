const postsReducer = (state = [], action) => {
  switch (action.type) {
    case "ADD_POST":
      return [action.post, ...state];
    case "EDIT_POST":
      return [
        ...state.map(post => {
          if (post.id === action.id) {
            return {
              ...post,
              ...action.updates
            };
          } else {
            return post;
          }
        })
      ];
    case "SET_POSTS":
      return [
        ...action.posts.map(post => {
          if (post.comments) {
            let comments = [];
            for (let i in post.comments) {
              comments.push(post.comments[i]);
            }
            post.comments = comments;
          }
          return post;
        })
      ];
    case "EDIT_POSTS_USER_PHOTO_URL":
      return [
        ...state.map(post => {
          if (post.uid == action.uid) {
            return {
              ...post,
              userPhotoURL: action.userPhotoURL
            };
          } else {
            return post;
          }
        })
      ];
    case "ADD_POST_COMMENT":
      return [
        ...state.map(post => {
          if (post.id == action.comment.postId) {
            if (!post.comments) {
              post.comments = [];
            }
            // post.comments = post.comments.push(action.comment);
            post.comments = [...post.comments, action.comment];
            return post;
          } else {
            return post;
          }
        })
      ];
    default:
      return state;
  }
};

export default postsReducer;
