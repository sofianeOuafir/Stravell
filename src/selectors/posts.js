const getVisiblePosts = (posts, { text, countryCode }) => {
  text = text.toLowerCase()
  return posts.filter(post => {
    const matchText =
      post.title.toLowerCase().includes(text) ||
      post.description.toLowerCase().includes(text);
    const matchCountry = countryCode === '' || post.countryCode === countryCode
    return matchText && matchCountry;
  });
};

export { getVisiblePosts };
