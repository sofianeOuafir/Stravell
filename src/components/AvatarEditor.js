import React, { Fragment } from "react";
import Editor from "react-avatar-editor";
import { connect } from "react-redux";
import uuid from "uuid";
import Avatar from "react-avatar";
import { IoMdCamera } from "react-icons/io";
import Modal from "react-modal";

import { uploadFile } from "./../aws/s3";
import { updateUserProfilePicture } from "./../queries/user";
import { updateProfilePicture } from "./../actions/auth";
import Loading from "./Loading";

class AvatarEditor extends React.Component {
  constructor(props) {
    super(props);
    const { userPhotoURL: currentImageUrl, uid } = props.user;
    this.state = {
      currentImageUrl,
      uploadedImage: null,
      uid,
      uploading: false,
      modalIsOpen: false
    };
  }

  closeModal = () => {
    this.setState(() => ({ modalIsOpen: false }));
  };

  onImageSave = () => {
    const { uid } = this.state;
    const { updateProfilePicture } = this.props;
    const canvas = this.editor.getImage();
    this.setState(() => ({ uploading: true, modalIsOpen: false }));

    canvas.toBlob(blob => {
      const file = new File([blob], "profile_picture", {
        type: blob.type
      });
      uploadFile({ file, location: `users/${uid}` })
        .then(data => {
          const { Location: newUserPhotoURL } = data;
          this.setState(() => ({
            uploading: false,
            currentImageUrl: `${newUserPhotoURL}?${new Date().getTime()}`
          }));

          updateUserProfilePicture({ uid, newUserPhotoURL });
          updateProfilePicture(`${newUserPhotoURL}?${new Date().getTime()}`);
        })
        .catch(e => {
          this.setState(() => ({ uploading: false }));
          alert(e);
        });
    });
  };

  onImageChange = e => {
    const uploadedImage = e.target.files[0];
    if (!uploadedImage) {
      return;
    }
    this.setState(() => ({ uploadedImage, modalIsOpen: true }));
  };

  onCancel = () => {
    this.setState(() => ({
      modalIsOpen: false
    }));
  };

  triggerFileDialog = () => {
    document.getElementById("imageInput").click();
  };

  setEditorRef = editor => (this.editor = editor);

  render() {
    const {
      uploadedImage,
      currentImageUrl,
      uploading,
      modalIsOpen
    } = this.state;
    return (
      <div
        className="relative flex align-items--center justify-content--center"
        style={{ height: 100, width: 100 }}
      >
        <Fragment>
          <Avatar
            className="absolute pointer"
            src={currentImageUrl}
            round={true}
            size={100}
            onClick={this.triggerFileDialog}
          />
          {uploading ? (
            <div className="absolute">
              <Loading size={40} />
            </div>
          ) : (
            <Fragment>
              <input
                id="imageInput"
                className="hide"
                type="file"
                onChange={this.onImageChange}
              />

              <IoMdCamera
                className="absolute pointer h2 text-white"
                onClick={this.triggerFileDialog}
              />
            </Fragment>
          )}
        </Fragment>

        <Modal
          isOpen={modalIsOpen}
          onRequestClose={this.closeModal}
          className="modal"
          contentLabel="Example Modal"
        >
          <p className="modal__title h3 favourite-font-weight">Edit Media</p>
          <Editor ref={this.setEditorRef} image={uploadedImage} />
          <div className="flex justify-content--between">
            <button className="button" onClick={this.onImageSave}>Save</button>
            <button className="button button--warm-peach" onClick={this.onCancel}>Cancel</button>
          </div>
        </Modal>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  updateProfilePicture: userPhotoURL =>
    dispatch(updateProfilePicture(userPhotoURL))
});

export default connect(
  null,
  mapDispatchToProps
)(AvatarEditor);
