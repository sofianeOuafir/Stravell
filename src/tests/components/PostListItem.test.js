import React from "react";
import { shallow } from "enzyme";
import { PostListItem } from "./../../components/PostListItem";
import posts from './../fixtures/posts';

let post;

beforeEach(() => {
  post = posts[0];
});

describe('the post contains a body and no URL has been provided', () => {
  test('should render correctly', () => {
    expect(post).toMatchObject({ provideURL: false, providedURL: "" });
    const wrapper = shallow(<PostListItem post={post} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('the post DO NOT contains a body and and a URL has been provided', () => {
  test('should render correctly', () => {
    const providedURL = "https://www.google.com";
    const provideURL = true;
    post.providedURL = providedURL;
    post.provideURL = provideURL;
    expect(post).toMatchObject({ provideURL, providedURL });
    const wrapper = shallow(<PostListItem post={post} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('the post belongs to the current user and props.editable is true', () => {
  test('should be possible to edit the post', () => {
    const wrapper = shallow(<PostListItem post={post} editable={true} isOwnedByCurrentUser={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('the post belongs DO NOT belongs to the current user and props.editable is true', () => {
  test('should NOT be possible to edit the post', () => {
    const wrapper = shallow(<PostListItem post={post} editable={true} isOwnedByCurrentUser={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('the post belongs to the current user and props.editable is false', () => {
  test('should NOT be possible to edit the post', () => {
    const wrapper = shallow(<PostListItem post={post} editable={false} isOwnedByCurrentUser={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});

test('should display correctly when post has an address', () => {
  const wrapper = shallow(<PostListItem post={post} />);
  expect(wrapper).toMatchSnapshot();
});

test('should display correctly when post does not have address', () => {
  post.address = "";
  const wrapper = shallow(<PostListItem post={post} />);
  expect(wrapper).toMatchSnapshot();
});

