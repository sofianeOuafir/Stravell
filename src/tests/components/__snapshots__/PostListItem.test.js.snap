// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correctly when post does not have address 1`] = `
<div
  className="post-list-item"
>
  <div
    className="post-list-item__content-container"
  >
    <a
      className="post-list-item__link"
      onClick={[Function]}
    >
      <img
        alt="Travel Image"
        className="post-list-item__image"
        src="https://stravell-production.s3.amazonaws.com/pictures/ea5f10a1-10c3-4cfd-a607-982678db7159/main"
      />
      <div
        className="post-list-item__title-description-container"
      >
        <h1
          className="post-list-item__title"
        >
          Don't Forget Where You Come From.
        </h1>
        <h2
          className="post-list-item__description"
        >
          Discovering the beauty of my country, my home.
        </h2>
      </div>
    </a>
  </div>
  <div>
    <div
      className="post-list-item__author-edit-container"
    >
      <PostAuthor
        authorName="Jenna Jacquelyn"
        authorPhotoURL="https://graph.facebook.com/10161147754845459/picture"
        authorUid="YYFEgQAdBuQIsox3ojncZtpSAmG3"
        avatarSize={40}
      />
    </div>
    <p>
      December 30th, 2018
    </p>
    <Link
      as="/p/show/don-t-forget-where-you-come-from/-LV-kv5VGQDljYtMPt9v"
      href="/post?id=-LV-kv5VGQDljYtMPt9v"
    >
      <a
        className="hide"
      />
    </Link>
  </div>
</div>
`;

exports[`should display correctly when post has an address 1`] = `
<div
  className="post-list-item"
>
  <div
    className="post-list-item__content-container"
  >
    <a
      className="post-list-item__link"
      onClick={[Function]}
    >
      <img
        alt="Travel Image"
        className="post-list-item__image"
        src="https://stravell-production.s3.amazonaws.com/pictures/ea5f10a1-10c3-4cfd-a607-982678db7159/main"
      />
      <div
        className="post-list-item__title-description-container"
      >
        <h1
          className="post-list-item__title"
        >
          Don't Forget Where You Come From.
        </h1>
        <h2
          className="post-list-item__description"
        >
          Discovering the beauty of my country, my home.
        </h2>
      </div>
    </a>
    <div
      className="post-list-item__address-container"
    >
      <Address
        address="Cannes"
        addressClassName="post-list-item__address"
        iconClassName="post-list-item__address-icon"
        placeId={1122313343}
      />
    </div>
  </div>
  <div>
    <div
      className="post-list-item__author-edit-container"
    >
      <PostAuthor
        authorName="Jenna Jacquelyn"
        authorPhotoURL="https://graph.facebook.com/10161147754845459/picture"
        authorUid="YYFEgQAdBuQIsox3ojncZtpSAmG3"
        avatarSize={40}
      />
    </div>
    <p>
      December 30th, 2018
    </p>
    <Link
      as="/p/show/don-t-forget-where-you-come-from/-LV-kv5VGQDljYtMPt9v"
      href="/post?id=-LV-kv5VGQDljYtMPt9v"
    >
      <a
        className="hide"
      />
    </Link>
  </div>
</div>
`;

exports[`the post belongs DO NOT belongs to the current user and props.editable is true should NOT be possible to edit the post 1`] = `
<div
  className="post-list-item"
>
  <div
    className="post-list-item__content-container"
  >
    <a
      className="post-list-item__link"
      onClick={[Function]}
    >
      <img
        alt="Travel Image"
        className="post-list-item__image"
        src="https://stravell-production.s3.amazonaws.com/pictures/ea5f10a1-10c3-4cfd-a607-982678db7159/main"
      />
      <div
        className="post-list-item__title-description-container"
      >
        <h1
          className="post-list-item__title"
        >
          Don't Forget Where You Come From.
        </h1>
        <h2
          className="post-list-item__description"
        >
          Discovering the beauty of my country, my home.
        </h2>
      </div>
    </a>
    <div
      className="post-list-item__address-container"
    >
      <Address
        address="Cannes"
        addressClassName="post-list-item__address"
        iconClassName="post-list-item__address-icon"
        placeId={1122313343}
      />
    </div>
  </div>
  <div>
    <div
      className="post-list-item__author-edit-container"
    >
      <PostAuthor
        authorName="Jenna Jacquelyn"
        authorPhotoURL="https://graph.facebook.com/10161147754845459/picture"
        authorUid="YYFEgQAdBuQIsox3ojncZtpSAmG3"
        avatarSize={40}
      />
    </div>
    <p>
      December 30th, 2018
    </p>
    <Link
      as="/p/show/don-t-forget-where-you-come-from/-LV-kv5VGQDljYtMPt9v"
      href="/post?id=-LV-kv5VGQDljYtMPt9v"
    >
      <a
        className="hide"
      />
    </Link>
  </div>
</div>
`;

exports[`the post belongs to the current user and props.editable is false should NOT be possible to edit the post 1`] = `
<div
  className="post-list-item"
>
  <div
    className="post-list-item__content-container"
  >
    <a
      className="post-list-item__link"
      onClick={[Function]}
    >
      <img
        alt="Travel Image"
        className="post-list-item__image"
        src="https://stravell-production.s3.amazonaws.com/pictures/ea5f10a1-10c3-4cfd-a607-982678db7159/main"
      />
      <div
        className="post-list-item__title-description-container"
      >
        <h1
          className="post-list-item__title"
        >
          Don't Forget Where You Come From.
        </h1>
        <h2
          className="post-list-item__description"
        >
          Discovering the beauty of my country, my home.
        </h2>
      </div>
    </a>
    <div
      className="post-list-item__address-container"
    >
      <Address
        address="Cannes"
        addressClassName="post-list-item__address"
        iconClassName="post-list-item__address-icon"
        placeId={1122313343}
      />
    </div>
  </div>
  <div>
    <div
      className="post-list-item__author-edit-container"
    >
      <PostAuthor
        authorName="Jenna Jacquelyn"
        authorPhotoURL="https://graph.facebook.com/10161147754845459/picture"
        authorUid="YYFEgQAdBuQIsox3ojncZtpSAmG3"
        avatarSize={40}
      />
    </div>
    <p>
      December 30th, 2018
    </p>
    <Link
      as="/p/show/don-t-forget-where-you-come-from/-LV-kv5VGQDljYtMPt9v"
      href="/post?id=-LV-kv5VGQDljYtMPt9v"
    >
      <a
        className="hide"
      />
    </Link>
  </div>
</div>
`;

exports[`the post belongs to the current user and props.editable is true should be possible to edit the post 1`] = `
<div
  className="post-list-item"
>
  <div
    className="post-list-item__content-container"
  >
    <a
      className="post-list-item__link"
      onClick={[Function]}
    >
      <img
        alt="Travel Image"
        className="post-list-item__image"
        src="https://stravell-production.s3.amazonaws.com/pictures/ea5f10a1-10c3-4cfd-a607-982678db7159/main"
      />
      <div
        className="post-list-item__title-description-container"
      >
        <h1
          className="post-list-item__title"
        >
          Don't Forget Where You Come From.
        </h1>
        <h2
          className="post-list-item__description"
        >
          Discovering the beauty of my country, my home.
        </h2>
      </div>
    </a>
    <div
      className="post-list-item__address-container"
    >
      <Address
        address="Cannes"
        addressClassName="post-list-item__address"
        iconClassName="post-list-item__address-icon"
        placeId={1122313343}
      />
    </div>
  </div>
  <div>
    <div
      className="post-list-item__author-edit-container"
    >
      <PostAuthor
        authorName="Jenna Jacquelyn"
        authorPhotoURL="https://graph.facebook.com/10161147754845459/picture"
        authorUid="YYFEgQAdBuQIsox3ojncZtpSAmG3"
        avatarSize={40}
      />
      <Link
        as="/p/edit/don-t-forget-where-you-come-from/-LV-kv5VGQDljYtMPt9v"
        href="/editPost?id=-LV-kv5VGQDljYtMPt9v"
        prefetch={true}
      >
        <a
          className="button"
        >
          Edit
        </a>
      </Link>
    </div>
    <p>
      December 30th, 2018
    </p>
    <Link
      as="/p/show/don-t-forget-where-you-come-from/-LV-kv5VGQDljYtMPt9v"
      href="/post?id=-LV-kv5VGQDljYtMPt9v"
    >
      <a
        className="hide"
      />
    </Link>
  </div>
</div>
`;

exports[`the post contains a body and no URL has been provided should render correctly 1`] = `
<div
  className="post-list-item"
>
  <div
    className="post-list-item__content-container"
  >
    <a
      className="post-list-item__link"
      onClick={[Function]}
    >
      <img
        alt="Travel Image"
        className="post-list-item__image"
        src="https://stravell-production.s3.amazonaws.com/pictures/ea5f10a1-10c3-4cfd-a607-982678db7159/main"
      />
      <div
        className="post-list-item__title-description-container"
      >
        <h1
          className="post-list-item__title"
        >
          Don't Forget Where You Come From.
        </h1>
        <h2
          className="post-list-item__description"
        >
          Discovering the beauty of my country, my home.
        </h2>
      </div>
    </a>
    <div
      className="post-list-item__address-container"
    >
      <Address
        address="Cannes"
        addressClassName="post-list-item__address"
        iconClassName="post-list-item__address-icon"
        placeId={1122313343}
      />
    </div>
  </div>
  <div>
    <div
      className="post-list-item__author-edit-container"
    >
      <PostAuthor
        authorName="Jenna Jacquelyn"
        authorPhotoURL="https://graph.facebook.com/10161147754845459/picture"
        authorUid="YYFEgQAdBuQIsox3ojncZtpSAmG3"
        avatarSize={40}
      />
    </div>
    <p>
      December 30th, 2018
    </p>
    <Link
      as="/p/show/don-t-forget-where-you-come-from/-LV-kv5VGQDljYtMPt9v"
      href="/post?id=-LV-kv5VGQDljYtMPt9v"
    >
      <a
        className="hide"
      />
    </Link>
  </div>
</div>
`;
